src/python/deepgnn/graph_engine/snark/meta_merger.py

```diff
@@ -48,97 +49,81 @@ def __init__(self, fs: fsspec.AbstractFileSystem, path: str) -> None:
         self._from_file()
 
     def _from_file(self):
-        with self.fs.open(self.path, "r") as meta:
-            self.version = meta.readline().strip()
-            if self.version[0] != "v":
-                raise RuntimeError(
-                    "First line in meta file should be version, please regenerate binary data"
-                )
-
-            self.node_count = int(meta.readline())
-            self.edge_count = int(meta.readline())
-            self.node_type_num = int(meta.readline())
-            self.edge_type_num = int(meta.readline())
-            self.node_feature_count = int(meta.readline())
-            self.edge_feature_count = int(meta.readline())
-            self.partition_count = int(meta.readline())
-            self.partition_ids = []
-            self.node_weights = [
-                [float(0)] * self.node_type_num for _ in range(self.partition_count)
-            ]
-            self.edge_weights = [
-                [float(0)] * self.edge_type_num for _ in range(self.partition_count)
-            ]
-            for p in range(self.partition_count):
-                self.partition_ids.append(int(meta.readline()))  # Partition id
-                for i in range(self.node_type_num):
-                    self.node_weights[p][i] = float(meta.readline())
-                for i in range(self.edge_type_num):
-                    self.edge_weights[p][i] = float(meta.readline())
-
-            self.node_count_per_type = []
-            self.edge_count_per_type = []
-
-            for _ in range(self.node_type_num):
-                self.node_count_per_type.append(int(meta.readline()))
-            for _ in range(self.edge_type_num):
-                self.edge_count_per_type.append(int(meta.readline()))
-
-    def get_partition(self):
+        with self.fs.open(self.path, "r") as file:
+            meta = json.load(file)  # type: ignore
+
+        self.version = meta["binary_data_version"]
+        if self.version[0] != "v":
+            raise RuntimeError(
+                "Meta file should contain binary_data_version, please regenerate binary data"
+            )
+
+        self.node_count = int(meta["node_count"])
```

Comment:
Can we drop some fields in the meta.json? E.g. `edge_count` can be probably inferred from all available partitions. This was harder to do with lists of numbers, but with structured format is way easier.

Reply:
Not in this PR.